python/paddle/fluid/compiler.py

```diff
@@ -101,9 +109,13 @@ class CompiledProgram(object):
             (potentially optimized before), it will be directly used for
             further optimizations. Note: graph is only supported when compiled
             with with_data_parallel option.
+        build_strategy(BuildStrategy): build_strategy is used to
+            build the graph with the specified options.
+            For more information, please refer to fluid.BuildStrategy.
+            Default None.
     """
 
-    def __init__(self, program_or_graph):
+    def __init__(self, program_or_graph, build_strategy=None):
```

Comment:
In this case, the `build_strategy` of `with_data_parallel()` will be overwritten if users pass `build_strategy` when creating `CompiledProgram`.

Reply:
what about rm the ‘build_strategy’ from with_data_parallel？

Reply:
Considering the compatibility problem, the original interface parameters can not be modified.

Reply:
Currently, the build_strategy can be set during creating CompiledProgram or calling with_data_parallel, and it may be confusing, but in the long run, we should set up build_strategy only when creating CompiledProgram, and exec_strategy should be deprecated.